[FEATURE] Enable full-width mode and dynamic height for redaccion.php markdown editor (#8145)

Ola-je opened the pull request:
… markdown editor (#8145)

# Description

This pull request implements the following improvements to the /redaccion.php view:

✅ Enables full-width display mode to maximize horizontal space.

✅ Adjusts the height of the markdown editor dynamically to fit the screen, improving usability for problem setters.

✅ Follows omegaUp UI conventions and clean layout principles.

This enhancement significantly improves the authoring experience, especially on high-resolution screens where too much space was previously left unused.

![Screenshot 2025-04-05 164807](https://github.com/user-attachments/assets/ad3e9aa3-ccae-4d1f-ad7f-7724632063fe)

Fixes: #8145

# Comments

- This is a beginner-friendly UI improvement. Please advise if further layout refinements are preferred.

- Let me know if I should submit a separate PR for additional styling or responsive improvements.

# Checklist:

- The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.

 - The changes were tested locally and function as expected.

-  (N/A) No new tests needed as this is a visual/layout enhancement.

-  (N/A) PR is under 200 lines — no need for multiple PRs.

pabo99 commented on frontend/www/redaccion.php:
```diff
@@ -3,13 +3,28 @@
+                'omegaupTitleStatementEditor'
             ),
             'payload' => [],
+            'scripts' => [
```
We already have a file that controls the entry point you'll be working on, which you can find in `webpack.config-frontend.js`:

https://github.com/omegaup/omegaup/blob/94b2f6f562dc9b2ff38887afda05ed0c469e2b55/webpack.config-frontend.js#L99C5-L99C73

So, the logic should go either in that file or directly in the component that renders the Markdown.

If you're not very familiar with Vue.js, let's do the following: let’s focus only on the change to make it full width for this PR, that way we can speed up the review process.

Does it sound good to you if you revert everything related to height adjustment, and we start working on that in a new PR instead?

pabo99 commented on frontend/www/redaccion.php:
```diff
@@ -3,13 +3,28 @@
 
 \OmegaUp\UITools::render(
-    /** @return array{templateProperties: array{fullWidth?: bool, hideFooterAndHeader?: bool, payload: array<string, mixed>, scripts?: list<string>, title: \OmegaUp\TranslationString}, entrypoint: string, inContest?: bool, navbarSection?: string} */
+    /** @return array{
+        templateProperties: array{
+            fullWidth?: bool,
+            hideFooterAndHeader?: bool,
+            payload: array<string, mixed>,
+            scripts?: list<string>,
+            title: \OmegaUp\TranslationString
+        },
+        entrypoint: string,
+        inContest?: bool,
+        navbarSection?: string
+    } */
```
The style in the declaration you proposed looks really good. However, we’ve standardized the content to be on a single line (though I’m not sure if PSALM complains about this). Maybe in the future, we can add a rule to the linter to enforce a consistent style for this.

For now, could you please revert this line?

Ola-je replied:
Thanks, @pabo99!

I didn't know about the standardized format, I edited it because I usually organize my codes like that. I will edit everything and make a new pr that only focuses on the full-width mode.

Ola-je commented:
Thanks again, @pabo99!

I've updated the PR so that it only includes the full-width mode for `redaccion.php` and reverted the dynamic height script.

The multiline PHPDoc has also been replaced with a single-line format to match the current code style.

I'll work on the height adjustment separately inside the Vue component and submit that as a new PR.

Let me know if this version is good to go! 🙂

pabo99 reviewed (approved):
Excellent! Thank you for your patience!

Let's move on to phase 3!

pabo99 commented:
@Ola-je could you please update your branch?
